mindsdb/integrations/handlers/huggingface_inference_api_handler/huggingface_inference_api_handler.py

```diff
@@ -0,0 +1,81 @@
+import os.path
+from typing import Optional, Dict
+
+import json
+import pandas as pd
+import requests
+
+from .config_parser import ConfigParser
+
+from mindsdb.integrations.libs.base import BaseMLEngine
+
+
+class HuggingFaceInferenceAPIHandler(BaseMLEngine):
+    """
+    Integration with the Hugging Face Inference API.
+    """
+
+    name = 'huggingface_inference_api'
+
+    def create(self, target: str, df: Optional[pd.DataFrame] = None, args: Optional[Dict] = None) -> None:
+        if 'using' not in args:
+            raise Exception("Hugging Face Inference engine requires a USING clause! Refer to its documentation for more details.")
+
+        config = ConfigParser(os.path.join(os.path.dirname(os.path.abspath(__file__)), 'config.yaml'))
+        config_args = config.get_config_dict()
+
+        self.model_storage.json_set('args', args)
+        self.model_storage.json_set('config_args', config_args)
+
+    def predict(self, df: Optional[pd.DataFrame] = None, args: Optional[Dict] = None) -> None:
+        args = self.model_storage.json_get('args')
+        config_args = self.model_storage.json_get('config_args')
+
+        inputs = self._parse_inputs(df, args['using']['inputs'], args['using']['task'])
+
+        response = self._query(
+            f"{config_args['BASE_URL']}/{config_args['TASK_MODEL_MAP'][args['using']['task']]}",
+            args['using']['api_key'],
```

Comment:
Much like in the OpenAI handler, would be nice to have multiple cases where this key can come from: USING args, but also env vars or at engine creation time.

Reply:
Done!